src/github/github.service.ts

```diff
@@ -0,0 +1,180 @@
+          clientId: string;
+          environment: string;
+          project: string;
+          service: string;
+          scmUrl: string;
+        };
+        const path = `${environment}/${project}/${service}`;
+        const kvData= await this.vaultService.getKv(VAULT_KV_APPS_MOUNT,path);
+        const value = kvData[`broker-jwt:${clientId}`];
+        if (!value) {
+          console.error(`No value found for client_id: ${clientId}`);
+          return;
+        }
+        if (scmUrl) {
+          const secretName = 'BROKER_JWT';
+          await this.updateSecret(scmUrl, secretName, value);
+          console.log('Secret updated successfully');
+        }
+        else
+          console.log('Service does not have Github repo URL to update:', service);
+      } catch(error) {
+        console.error('Failed to retrieve KV data or update GitHub secret:', error);
+      }
+    });
+  }
+
+  // Generate JWT
+  private generateJWT():string {
+    const payload = {
+      iat: Math.floor(Date.now() / 1000) - 60,
+      exp: Math.floor(Date.now() / 1000) + (2 * 60), // JWT expires in 2 minutes
+      iss: this.clientId,
+    };
+    return jwt.sign(payload, this.privateKey, { algorithm: 'RS256' });
+  }
+
+  private async getInstallationId(owner: string, repo: string, token: string): Promise<string> {
+    try {
+      const response = await this.axiosInstance.get(
+        `/repos/${owner}/${repo}/installation`,
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      );
+      return response.data.id;
+    } catch (error)  {
+      console.error("catch error on make API call on get Installation ID", error);
+      throw new Error('Failed to get installation id.');
+    }
+  }
+
+  public async getInstallationAccessToken(owner: string, repo: string): Promise<string> {
+    const token = this.generateJWT();
+    try {
+      const InstallationId = await this.getInstallationId(owner, repo, token);
+      const response = await this.axiosInstance.post(
+        `/app/installations/${InstallationId}/access_tokens`,
+        {},
+        {
+          headers: {
+            Authorization: `Bearer ${token}`,
+          },
+        }
+      );
+      return response.data.token;
+    } catch (error)  {
+      console.error("catch error on make API call on generate access token", error);
+      throw new Error('Failed to get access token.');
+    }
+  }
+
+
+  public getOwnerAndRepoFromUrl(repoUrl: string): { owner: string; repo: string } {
+    const regex = /github\.com[:/](.+?)\/(.+?)(\.git)?$/;
+    const match = repoUrl.match(regex);
+    if (match && match.length >= 3) {
+      return { owner: match[1], repo: match[2] };
+    }
+    throw new Error('Invalid GitHub URL');
+  }
+
+  // Implement encryption logic
+  private async encryptSecret(secretValue: string, publicKey: string): Promise<string> {
+
+    try {
+      await sodium.ready;
```

Comment:
convert to builtin crypto library. Hint! chagtpt is your friend:
rewrite using nodejs api: const encryptedUint8Array = sodium.crypto_box_seal(secretUint8Array, publicKeyUint8Array);